src/riak_kv_leveled_backend.erl

```diff
@@ -616,9 +616,7 @@ status(_State) ->
 
 %% @doc Get an estimate of the data_size for this leveled backend
 -spec data_size(state()) ->
-    undefined |
-    {non_neg_integer(), objects} |
-    {fun(() -> {non_neg_integer(), objects}), dynamic}.
+    {fun(() -> {non_neg_integer(), objects}), async}.
```

Comment:
This is not the same type as used in `riak_kv_backend`. It is a proper subtype, so that should be fine.